frontend/packages/ceph-storage-plugin/src/components/create-storage-system/payloads.ts

```diff
@@ -139,3 +141,29 @@ export const createExternalSubSystem = async (subSystemPayloads: Payload[]) => {
     throw err;
   }
 };
+
+const validateCRDReadiness = (crd: K8sResourceKind, plural: string) =>
+  crd?.status?.acceptedNames?.plural === plural;
+
+export const waitforCRD = async (name: string, plural: string, maxAttempts = 15) => {
+  let attempts = 0;
+
+  const pollCRD = async (resolve, reject) => {
+    try {
+      attempts++;
+      const crd = await k8sGet(CustomResourceDefinitionModel, name);
+      const isCrdReady = validateCRDReadiness(crd, plural);
+      if (isCrdReady) return resolve(crd);
```

Comment:
If this returned CRD object ever used? If not please don't return it.

Reply:
This is a promise.
It should be resolved. Added return due to eslint errors.
Not having any effect. Also returing if in case this value need to be used.

Reply:
You can always return null.

Reply:
k8sWatch only work for the resources that are present in your cluster. 
Resolving by crd was only there because I wanted this function to be useful. Thats the only idea. Its upto usage to use crd or not

Reply:
`CustomResourceDefinition` **model** should always be present. Based on this we can look for a particular resource of kind CRD IIUC.

Reply:
CRD are injected by operator. Not always present like k8s resource

Reply:
CRD model is always present in k8s. PTAL https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#customresourcedefinition-v1-apiextensions-k8s-io

Reply:
@bipuladh I think you need to go through the PR.
We are taking about pull a crd resource type - oc get crd <my_crd>/ This Pr is not about just the CRD.